src/flowMC/strategy/adapt_step_size.py

```
@@ -150,3 +150,137 @@ def __call__(
+        assert isinstance(kernel, ProposalBase), (
+            f"Resource '{self.kernel_name}' must be a ProposalBase, got {type(kernel)}"
+        )
+        assert hasattr(kernel, "get_effective_dim_profile") and hasattr(
+            kernel, "apply_per_dim_scaling"
+        ), (
+            f"Kernel '{self.kernel_name}' must implement get_effective_dim_profile() "
+            f"and apply_per_dim_scaling(). Got {type(kernel)}."
+        )
+
+        # Current per-dim profile from the kernel (geomean=1, kernel-specific)
+        current_profile = getattr(kernel, "get_effective_dim_profile")()
```

Ruff (0.15.13)

[warning] Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)